client/src/components/NewIngredient.js

```diff
@@ -0,0 +1,194 @@
+    let newErrors = {};
+    if (name.trim() == "") {
+      newErrors = {
+        ...newErrors,
+        name: "is required",
+      };
+      foundError = true
+    }
+
+    if (amount.trim() == "") {
+      newErrors = {
+        ...newErrors,
+        amount: "is required",
+      };
+      foundError = true
+    }
+
+
+    if (unit.trim() == "") {
+      newErrors = {
+        ...newErrors,
+        unit: "is required",
+      };
+      foundError = true
+    }
+
+    if (unit.trim() == "other") {
+      newErrors = {
+        ...newErrors,
+        unit: "please enter units"
+      }
+    }
+
+    setErrors(newErrors);
+    return foundError
+  }
+
+  const handleSubmit = async (event) => {
+    event.preventDefault()
+    const foundError = validateData(formData)
+    if (!foundError) {
+      const formPayload = {
+        newIngredient: { ...formData },
+        userId: props.user.id
+      }
+      try {
+        const response = await fetch(`/api/v1/ingredients`, {
+          method: "POST",
+          headers: new Headers({
+            "Content-Type": "application/json"
+          }),
+          body: JSON.stringify(formPayload)
+        })
+        if(!response.ok) {
+          if(response.status === 422) {
+            const body = await response.json()
+            const newErrors = translateServerErrors(body.errors)
+            return setServerErrors(newErrors)
+          } else {
+            const errorMessage = `${response.status}: (${response.statusText})`
+            const error = new Error(errorMessage)
+            throw(error)
+          }
+        } else {
+          setShouldRedirect(true)
+        }
+      } catch(error) {
+        console.error(`Error in Fetch: ${error.message}`)
+      }
+    }
+  }
+
+  const handleChange = (event) => {
+    setFormData({
+      ...formData,
+      [event.currentTarget.name]: event.currentTarget.value
+    })
+  }
+
+  const handleSelect = (event) => {
+    setFormData({
+      ...formData,
+      [event.currentTarget.name]: event.currentTarget.value
+    })
+    if (event.currentTarget.value === "other") {
+      setShowOther(true)
+    } else {
+      setShowOther(false)
+    }
+  }
+
+  let otherField
+  if (showOther) {
+    otherField = (
+      <div>
+        <label htmlFor="other">Other: </label>
+          <input 
+            type="text" 
+            id="other" 
+            name="unit"
+            value={formData.unit}
+            onChange={handleChange}
+            />
+      </div>
+    )
+  }
+
+  if (shouldRedirect) {
+    location.href="/ingredients"
+  }
+
+  return(
+    <div>
+      <h2>Add Your New Ingredient</h2>
+      <form className="callout secondary" onSubmit={handleSubmit}>
+        <ErrorList errors={serverErrors} />
+        <label htmlFor="name">Name: </label>
+        <input 
+          type="text" 
+          id="name" 
+          name="name" 
+          value={formData.name}
+          onChange={handleChange}
+          />
```

Comment:
this does need to be a finish you do now, but right here would be a GREAT place to implement a auto-fill form input. Exactly like your search bar in group projects, as someone types we can make a fetch request to search for ingredients that are in the database. if results come back, we can show these in a small dropdown. If no results, the dropdown can say "no results, add original ingreident" or something